lib/manager/npm/post-update/yarn.spec.ts

```diff
@@ -242,85 +246,62 @@ describe('manager/npm/post-update/yarn', () => {
-            resolve(
-              'yarn-offline-mirror "./packages-cache"\nyarn-path "./.yarn/cli.js"\n'
-            )
-          );
-        }
-        return new Promise<string>((resolve) => resolve(''));
-      });
-      expect(await _yarnHelper.checkYarnrc('/tmp/renovate')).toEqual({
+      Fixtures.mock(
+        {
+          '.yarn/cli.js': '',
+          '/tmp/renovate/.yarnrc':
+            'yarn-offline-mirror "./packages-cache"\nyarn-path "/.yarn/cli.js"\n',
+        },
+        '/'
+      );
+      expect(await yarnHelper.checkYarnrc('/tmp/renovate')).toEqual({
         offlineMirror: true,
-        yarnPath: './.yarn/cli.js',
+        yarnPath: '/.yarn/cli.js',
       });
     });
 
     it('returns no offline mirror and unquoted yarn path', async () => {
-      fs.exists.mockImplementation((path) => {
-        if (path === './.yarn/cli.js') {
-          return new Promise<boolean>((resolve) => resolve(true));
-        }
-        return new Promise<boolean>((resolve) => resolve(false));
-      });
-      fs.readFile.mockImplementation((filename, encoding) => {
-        if (filename.endsWith('.yarnrc')) {
-          return new Promise<string>((resolve) =>
-            resolve('yarn-path ./.yarn/cli.js\n')
-          );
-        }
-        return new Promise<string>((resolve) => resolve(''));
-      });
-      expect(await _yarnHelper.checkYarnrc('/tmp/renovate')).toEqual({
+      Fixtures.mock(
+        {
+          '.yarn/cli.js': '',
+          '/tmp/renovate/.yarnrc': 'yarn-path /.yarn/cli.js\n',
+        },
+        '/'
+      );
+      expect(await yarnHelper.checkYarnrc('/tmp/renovate')).toEqual({
         offlineMirror: false,
-        yarnPath: './.yarn/cli.js',
+        yarnPath: '/.yarn/cli.js',
       });
     });
 
     it('returns offline mirror and no yarn path for non-existant yarn-path binary', async () => {
-      let yarnrcContents = 'yarn-path ./.yarn/cli.js\n';
-      fs.writeFile.mockImplementation((filename, fileContents) => {
-        if (filename.endsWith('.yarnrc')) {
-          yarnrcContents = fileContents;
-        }
-        return new Promise<void>((resolve) => resolve());
-      });
-      fs.readFile.mockImplementation((filename, encoding) => {
-        if (filename.endsWith('.yarnrc')) {
-          return new Promise<string>((resolve) => resolve(yarnrcContents));
-        }
-        return new Promise<string>((resolve) => resolve(''));
-      });
-      const { offlineMirror, yarnPath } = await _yarnHelper.checkYarnrc(
+      Fixtures.mock(
+        {
+          '.yarnrc': 'yarn-path ./.yarn/cli.js\n',
+        },
+        '/tmp/renovate'
+      );
+      const { offlineMirror, yarnPath } = await yarnHelper.checkYarnrc(
         '/tmp/renovate'
       );
       expect(offlineMirror).toBeFalse();
       expect(yarnPath).toBeNull();
-      expect(yarnrcContents).not.toContain('yarn-path');
+      expect(Fixtures.toJSON('/tmp/renovate/.yarnrc')).not.toContain(
```

Comment:
Not sure this assertion works as expected. If I'm correct, previously it was testing string to contain `yarn-path` substring. Now, return value for `Fixtures.toJSON('/tmp/renovate/.yarnrc')` is `{ '/tmp/renovate/.yarnrc': '\n' }`, and `toContain` doesn't work well for objects.

Reply:
Yeah, and i don't really know what the `toJSON` function is doing. So see comments below

Reply:
Changed to `expect(Fixtures.toJSON()['/tmp/renovate/.yarnrc']).toBe('\n');`